lib/pure/includes/osenv.nim

```diff
@@ -175,6 +181,32 @@ proc putEnv*(key, val: string) {.tags: [WriteEnvEffect].} =
       if c_putenv(environment[indx]) != 0'i32:
         raiseOSError(osLastError())
 
+proc unsetEnv*(key: string) {.tags: [WriteEnvEffect].} =
+  ## Removes the `environment variable`:idx: named `key`.
+  ## If an error occurs, `OSError` is raised.
+  ##
+  ## See also:ven
+  ## * `getEnv proc <#getEnv,string,string>`_
+  ## * `existsEnv proc <#existsEnv,string>`_
+  ## * `putEnv proc <#putEnv,string,string>`_
+  ## * `envPairs iterator <#envPairs.i>`_
+  when nimvm:
+    discard "built into the compiler"
```

Comment:
Whats the point of this line, better put that on the RST comment above, at least people will see it:

```rst
## Note: For NimScript target, ``unsetEnv`` functionality is built into the Nim compiler itlself.
```

Reply:
Actually, I'm not sure if `unsetEnv` would work in nimscript. @Araq Do I need to do anything special to make it work there too?

Reply:
I dont think you can use `"<stdlib.h>"` directly like that on NimScript, needs testin' tho. :thinking:

Reply:
I see:

https://github.com/nim-lang/Nim/blob/7cf9b522b5d2735b5279e9c2bdcade5c177868f8/lib/system/nimscript.nim#L116-L118

I don't understand how that "builtin" works .. do I just copy/paste that and add `unsetEnv` to nimscript.nim too?